src/library/scala/runtime/ModuleSerializationProxy.java

```diff
@@ -48,4 +46,56 @@ public ModuleSerializationProxy(Class<?> moduleClass) {
     private Object readResolve() {
         return instances.get(moduleClass);
     }
+
+    private static boolean checkClassValueAvailability() {
+        try {
+            Class.forName("java.lang.ClassValue", false, Object.class.getClassLoader());
+            return true;
+        } catch (ClassNotFoundException e) {
+            return false;
+        }
+    }
+
+    private interface ClassValueInterface<T> {
```

Comment:
Do we have to have a Java version and a Scala version?

Reply:
Made an attempt to convert ModuleSerializationProxy.java to Scala but I am not very sure if this will cause any kind of ill side-effects for not knowing what exactly it does.